### PR TITLE
Use current balance in calculator once user logs in

### PR DESCRIPTION
**Pull request by AlaaElattar**
### Description
- Once user logs in, the current balance will be used in calculator.

### Related Issues

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1863

[Screencast from 01-04-2024 10:36:46 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/80e849ef-e6be-49a0-b83f-cb5739e8a356)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by mohamedamer453 — COMMENTED** [review]
- The first time after login the use balance didn't work as expected and it was enabled but it didn't read my balance.

[Screencast from 04-01-24 12:33:12.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/7c5d2bc6-4e2b-449e-abb8-8603f8658cdf)

**Comment by AlaaElattar** [comment]
> * The first time after login the use balance didn't work as expected and it was enabled but it didn't read my balance.
> 
>  Screencast.from.04-01-24.12.33.12.webm

should be fixed

File: packages/playground/src/calculator/resource_pricing.vue [hunk 1]
```diff
@@ -230,32 +231,53 @@ const calculator = computed(() => {
...
     balance.value = 0;
   }
 });
 
+watch(
+  hasActiveProfile,
+  async newVal => {
+    if (newVal) {
+      currentbalance.value = true;
+      const profile = profileManager.profile!;
```
Comment (0oM4R) [inline packages/playground/src/calculator/resource_pricing.vue:270]: can we add something that indicate that the balance is loading ?

[Screencast from 04 ينا, 2024 EET 03:54:58 م.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/838ee21c-d948-4cfb-bcc5-00b96bf9082c)

Comment (AlaaElattar) [reply packages/playground/src/calculator/resource_pricing.vue:275]: [Screencast from 01-04-2024 04:12:43 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/54944491/50c55aa0-7880-472d-9897-a1230ec5edda)

File: packages/playground/src/dashboard/transfer_view.vue [hunk 2]
```diff
@@ -232,7 +232,6 @@ async function submitFormTwinID() {
     if (twinDetails != null) {
       loadingTwinIDTransfer.value = true;
       await transfer(twinDetails);
-      loadingTwinIDTransfer.value = false;
```
Comment (0oM4R) [inline packages/playground/src/dashboard/transfer_view.vue]: is this related ?

Comment (AlaaElattar) [reply packages/playground/src/dashboard/transfer_view.vue]: no i returned it, it was by mistake when i was merging conflicts.

File: packages/playground/src/calculator/resource_pricing.vue [hunk 3]
```diff
@@ -209,6 +211,9 @@ const isCertified = ref<boolean>(false);
 const ipv4 = ref<boolean>(false);
 const currentbalance = ref<boolean>(false);
 
+const profileManagerController = useProfileManagerController();
+const freeBalance = computed(() => profileManagerController.balance.value?.free ?? 0);
```
Comment (0oM4R) [inline packages/playground/src/calculator/resource_pricing.vue:215]: i suggest to set it to fixed 2 decimals 

```suggestion
const freeBalance = computed(() => (profileManagerController.balance.value?.free)?.toFixed(2) ?? 0);
```

image

File: packages/playground/src/calculator/resource_pricing.vue [hunk 4]
```diff
@@ -230,32 +235,61 @@ const calculator = computed(() => {
...
-  let pkgs: any;
-  if (!valid.value.error) {
-    if (currentbalance.value && hasActiveProfile.value && gridStore.grid) {
-      const accountBalance = await gridStore.grid?.balance.getMyBalance();
-      balance.value = accountBalance?.free;
+watch(
+  [CRU, MRU, SRU, HRU, balance, isCertified, ipv4, currentbalance, hasActiveProfile],
+  async () => {
+    let pkgs: any;
+    if (!valid.value.error) {
```
Comment (0oM4R) [inline packages/playground/src/calculator/resource_pricing.vue:242]: we can avoid those errors
image
```suggestion
    if (!valid.value?.error) {
```
